controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -99,11 +137,414 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases(ctx context.Contex
+				"osd": "allow rw tag cephfs metadata=*",
+			},
+		}
+		return nil
+	})
+
+	if err != nil {
+		r.log.Error(
+			err,
+			"Failed to update CephClient.",
+			"CephClient",
+			klog.KRef(r.cephClientCephFSProvisioner.Namespace, r.cephClientCephFSProvisioner.Name),
+		)
+		return err
+	}
+
+	phase := ""
+	if r.cephClientCephFSProvisioner.Status != nil {
+		phase = string(r.cephClientCephFSProvisioner.Status.Phase)
+	}
+
+	r.setCephResourceStatus(r.cephClientCephFSProvisioner.Name, "CephClient", phase, nil)
+
+	return nil
+}
+
+func (r *StorageClassClaimReconciler) reconcileCephClientCephFSNode() error {
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientCephFSNode, func() error {
+		if err := r.own(r.cephClientCephFSNode); err != nil {
+			return err
+		}
+
+		addStorageRelatedAnnotations(r.cephClientCephFSNode, r.storageClassClaim.Name, "cephfs", "node")
+		r.cephClientCephFSNode.Spec = rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "allow r",
+				"mgr": "allow rw",
+				"osd": "allow rw tag cephfs *=*",
+				"mds": fmt.Sprintf("allow rw path=/volumes/%s", r.cephFilesystemSubVolumeGroup.Name),
+			},
+		}
+		return nil
+	})
+
+	if err != nil {
+		r.log.Error(
+			err,
+			"Failed to update CephClient.",
+			"CephClient",
+			klog.KRef(r.cephClientCephFSNode.Namespace, r.cephClientCephFSNode.Name),
+		)
+		return err
+	}
+
+	phase := ""
+	if r.cephClientCephFSNode.Status != nil {
+		phase = string(r.cephClientCephFSNode.Status.Phase)
+	}
+
+	r.setCephResourceStatus(r.cephClientCephFSNode.Name, "CephClient", phase, nil)
+
+	return nil
+}
+
+func (r *StorageClassClaimReconciler) setCephResourceStatus(name string, kind string, phase string, cephClients map[string]string) {
+
+	cephResourceSpec := r.cephResourcesByName[name]
+
+	if cephResourceSpec == nil {
+		cephResourceSpec = &ocsv1alpha1.CephResourcesSpec{
+			Name:        name,
+			Kind:        kind,
+			CephClients: cephClients,
+		}
+		r.storageClassClaim.Status.CephResources = append(r.storageClassClaim.Status.CephResources, cephResourceSpec)
+		r.cephResourcesByName[name] = cephResourceSpec
+	}
+
+	cephResourceSpec.Phase = phase
+}
+
+func addStorageRelatedAnnotations(obj client.Object, storageClassClaimName, storageClaim, cephUserType string) {
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		annotations = map[string]string{}
+		obj.SetAnnotations(annotations)
+	}
+
+	annotations[StorageClassClaimAnnotation] = storageClassClaimName
+	annotations[controllers.StorageClaimAnnotation] = storageClaim
+	annotations[controllers.StorageCephUserTypeAnnotation] = cephUserType
+}
+
+func (r *StorageClassClaimReconciler) own(resource metav1.Object) error {
+	// Ensure storageClassClaim ownership on a resource
+	return ctrl.SetControllerReference(r.storageClassClaim, resource, r.Scheme)
+}
+
 func (r *StorageClassClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
```

Comment:
> First, please move this function to the top of the file as it is the first thing that will be called from this file. 

ack

>Second, you are missing a bunch of `Own()` to initiate a watch on owned resources from a specific type. Without this, the reconciler will not be triggered when an "owned" resource is changed.

will make change soon

Reply:
@subhamkrai The `r.own` you are running inside `CreateOrUpdate` is adding an owner reference to resources, it will not automatically create an active watch in these resources! You need to have an `Own` watch in SetupWithManager to achieve that.

Please take a look at how it is done in other controllers, for example in the storageconsumer contoller here: 
https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storageconsumer/storageconsumer_controller.go#L671-L678

Reply:
added `own` for `blockpool`, `filesystem`, `cephClient`